sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs

```diff
@@ -4,18 +4,23 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel;
+using System.Text.Json.Serialization;
 using Azure.Core;
 
 namespace Azure.Data.AppConfiguration
 {
     /// <summary>
     /// A setting, defined by a unique combination of a key and label.
     /// </summary>
+    [JsonConverter(typeof(ConfigurationSettingJsonConverter))]
     public sealed class ConfigurationSetting
     {
         private IDictionary<string, string> _tags;
 
-        internal ConfigurationSetting()
+        /// <summary>
+        /// Creates a configuration setting and sets the values from the passed in parameter to this setting.
+        /// </summary>
+        public ConfigurationSetting()
```

Comment:
Let's not add a public parameterless constructor if we can help it.  It looks like identity was able to work around this... can we follow their example?

https://github.com/Azure/azure-sdk-for-net/blob/0bd6e12f67244e81bd9c6dcac17fe889dece5155/sdk/identity/Azure.Identity/src/AadIdentityClient.cs#L279

Reply:
It is actually not required. It is a left from the earlier experiment.